res/controllers/Pioneer-DDJ-200-scripts.js

```diff
@@ -19,11 +19,36 @@ DDJ200.init = function() {
 
         var vgroup = "[Channel" + i + "]";
 
-        // run onTrackLoad after every track load to set LEDs accordingly
+        // run updateDeckLeds after every track load to set LEDs accordingly
         engine.makeConnection(vgroup, "track_loaded", function(ch, vgroup) {
-            DDJ200.onTrackLoad(ch, vgroup);
+            DDJ200.updateDeckLeds(vgroup);
         });
 
+        // run switchPlayLED after play/pause track to set LEDs accordingly
+        engine.makeConnection(vgroup, "play", function(ch, vgroup) {
+            var vDeckNo = script.deckFromGroup(vgroup);
+            var d = (vDeckNo % 2) ? 0 : 1;
+            DDJ200.switchPlayLED(d, ch);
+        });
+
+        // run switchSyncLED after sync toogle to set LEDs accordingly
+        engine.makeConnection(vgroup, "sync_enabled", function(ch, vgroup) {
+            var vDeckNo = script.deckFromGroup(vgroup);
+            var d = (vDeckNo % 2) ? 0 : 1;
+            DDJ200.switchSyncLED(d, ch);
+        });
+
+        // listen to changes on hotcues
+        for (var j = 1; j <= 8; j++) {
+            // run switchPadLED after every hotcue update
+            engine.makeConnection(vgroup, "hotcue_" + j + "_enabled", function(ch, vgroup, control) {
+                var pad = Number(control.split("_")[1]);
```

Comment:
```suggestion
                var pad = j;
```
Should have the same effect, but please test.

Reply:
This don't work cause the callback function is in another context.  As the function executes outside the for loop, `j` will be always 8.

I could write this instead:

```javascript
            engine.makeConnection(vgroup, "hotcue_" + j + "_enabled", function (pad) {
                return function(ch, vgroup, control) {
                    var vDeckNo = script.deckFromGroup(vgroup);
                    var d = (vDeckNo % 2) ? 0 : 1;           // d = deckNo - 1
                    DDJ200.switchPadLED(d, pad, ch);
                };
            }(j));
```

But I think the code readability would be compromised.

Reply:
I guess you are right: [Creating closures in loops: A common mistake](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures#creating_closures_in_loops_a_common_mistake)

Reply:
Interesting, I figured `j` was copied in the loop because its just number. Both of your solutions are fine. You can keep it as is if you like.

Reply:
@Swiftb0y If I had to use JavaScript I would fail miserably. Looked it up out of curiosity and was surprised, unpleasantly of course.